### PR TITLE
Add a blocking thread on the message queue

### PR DESCRIPTION
**Pull request by 31Nathan**
Add a function for the blocking thread
Read the message queue and transform the command into a Pizza Object

### CODE REVIEW

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
See you tomorrow

File: src/Core/Kitchen/Kitchen.cpp [hunk 1]
```diff
@@ -9,18 +9,136 @@
 
 #include "Kitchen.hpp"
 #include <ctime>
+#include <iostream>
```
Comment (osvegn) [inline src/Core/Kitchen/Kitchen.cpp:12]: usefull ?

File: src/Core/Kitchen/Kitchen.hpp [hunk 2]
```diff
@@ -56,7 +57,21 @@ namespace plazza {
...
+            static int getQuantityFromFullCommand(const std::string &FullCommand);
+
+            /// \brief get the necessary ingredients from the pizza type
+            static void getIngredientsFromPizzaType(Pizza &toCook, PizzaType type);
         private:
+            /// \brief Function getting the command and give it in the job queue
+            static void _receptCook(Kitchen *obj);
+
             /// \brief Main function instancied in each thread
             static void _Cook(Kitchen *obj);
```
Comment (osvegn) [inline src/Core/Kitchen/Kitchen.hpp:76]: need more documentation

File: src/Core/Plazza.hpp [hunk 3]
```diff
@@ -44,6 +47,10 @@ namespace plazza {
             /// \brief Handle the receive input
             /// \return true if the input is valid, false otherwise
             bool _handleInput(const std::string &input);
+
+            void _sendCommand(const InputParser &command);
+
+            std::map<pid_t, std::shared_ptr<MessageQueue>> _kitchenQueues;
```
Comment (osvegn) [inline src/Core/Plazza.hpp:53]: need documentation

**Review by Cavonstavant — APPROVED** [review]

**Review by osvegn — CHANGES_REQUESTED** [review]
Need minor changes

**Review by osvegn — APPROVED** [review]
LGTM

**Pull request merged**